spec/services/ccms/case_add_requestor_xml_blocks_spec.rb

```diff
@@ -846,6 +849,189 @@ module CCMS # rubocop:disable Metrics/ModuleLength
+          expect(block).to be_present
+          expect(block).to have_response_type 'text'
+          expect(block).to have_response_value 'MULTIPLE'
+        end
+
+        it 'NEW_OR_EXISTING should be hard coded to NEW' do
+          attributes = [
+            [:proceeding, 'NEW_OR_EXISTING'],
+            [:proceeding_merits, 'NEW_OR_EXISTING']
+          ]
+          attributes.each do |entity_attribute_pair|
+            entity, attribute = entity_attribute_pair
+            block = XmlExtractor.call(xml, entity, attribute)
+            expect(block).to be_present
+            expect(block).to have_response_type 'text'
+            expect(block).to have_response_value 'NEW'
+          end
+        end
+
+        it 'RELATIONSHIP_TO_CASE should be hard coded to OPP' do
+          block = XmlExtractor.call(xml, :opponent, 'RELATIONSHIP_TO_CASE')
+          expect(block).to be_present
+          expect(block).to have_response_type 'text'
+          expect(block).to have_response_value 'OPP'
+        end
+
+        it 'OTHER_PARTY_TYPE should be hard coded to PERSON' do
+          block = XmlExtractor.call(xml, :opponent, 'OTHER_PARTY_TYPE')
+          expect(block).to be_present
+          expect(block).to have_response_type 'text'
+          expect(block).to have_response_value 'PERSON'
+        end
+
+        it 'POA_OR_BILL_FLAG should be hard coded to N/A' do
+          block = XmlExtractor.call(xml, :global_means, 'POA_OR_BILL_FLAG')
+          expect(block).to be_present
+          expect(block).to have_response_type 'text'
+          expect(block).to have_response_value 'N/A'
+        end
+
+        it 'LAR_INPUT_T_1WP2_8A should be hard coded correctly' do
+          block = XmlExtractor.call(xml, :global_means, 'LAR_INPUT_T_1WP2_8A')
+          expect(block).to be_present
+          expect(block).to have_response_type 'text'
+          expect(block).to have_response_value 'Apply Service Application. See uploaded means report in CCMS'
+        end
+
+        it 'should be hard coded with the correct notification' do
+          attributes = [
+            [:global_merits, 'REASON_APPLYING_FHH_LR'],
+            [:global_merits, 'REASON_NO_ATTEMPT_TO_SETTLE'],
+            [:global_merits, 'REASON_SEPARATE_REP_REQ'],
+            [:global_merits, 'INJ_REASON_POLICE_NOT_NOTIFIED'],
+            [:proceeding_merits, 'INJ_RECENT_INCIDENT_DETAIL']
+          ]
+          attributes.each do |entity_attribute_pair|
+            entity, attribute = entity_attribute_pair
+            block = XmlExtractor.call(xml, entity, attribute)
+            expect(block).to be_present
+            expect(block).to have_response_type 'text'
+            # Issue with this as it is not matching the text correctly, seems to be due to length of the string
+            # expect(block).to have_response_value 'Apply Service application. See provider statement of case and report uploaded as evidence'
```

Comment:
One problem with defining spec expectations within a loop, is that in their standard form, it is very difficult to determine which iteration of the loop failed. Therefore, I would strongly recommend that you use a [custom message](https://relishapp.com/rspec/rspec-expectations/v/3-8/docs/customized-message) inside loops so that the current value of entity being tested can be identified.

Here, that would be something like:
```ruby
expect(block).to have_response_value 'Apply Service application. See provider statement of case and report uploaded as evidence', "expected #{entity} #{attribute} to have matching value"
```

Reply:
```have_response_value``` already does this